### DISCUSSION TITLE
dotnet test --collect:"XPlat Code Coverage" not considering Api project

### DISCUSSION THREAD

**Discussion by LokanathCalyx** [coverlet-coverage/coverlet#1411] [Q&A]
I am trying to generate Code coverage report in 2 ways.

Using Fine Code Coverage. There i see all 3 projects of me covered & its % of coverage.
Using Azure devops Pipeline where i use DotNetCoreCLI@2& PublishCodeCoverageResults@1 to build & generate the report. But the issue is API project is not considered in the report. both in trx & xml file.
So i tried MSBuild commands to repro the issue. To run the test cases & to build the xml file i am using below command.

`dotnet test --collect:"XPlat Code Coverage"   `
After that to publish the coverage using below command.

`dotnet test /p:CollectCoverage=true`
Issue:- In above command also , i don't see my Api Project is getting into report, where as i see the API layer unit test are running. tried with [include] but seems not working.

Below is my project structure( using clean code architecture/Onion architecture)

![image](https://user-images.githubusercontent.com/89831094/202644474-f6957840-bf4c-439d-96b9-c9e65ac698a8.png)

**Comment by Bertk** [comment]
Please use coverlet.console for integrating testing (see [Code Coverage for integration tests and end-to-end tests](https://github.com/coverlet-coverage/coverlet/blob/893dcc8fb6e9dbc76c159620af66134fd8c2414e/Documentation/GlobalTool.md#code-coverage-for-integration-tests-and-end-to-end-tests))
